search-commons/src/main/java/no/unit/nva/search2/common/QueryTools.java

```diff
@@ -0,0 +1,235 @@
+package no.unit.nva.search2.common;
+
+import static java.util.Objects.nonNull;
+import static no.unit.nva.search2.constant.ErrorMessages.OPERATOR_NOT_SUPPORTED;
+import static no.unit.nva.search2.constant.Words.AMPERSAND;
+import static no.unit.nva.search2.constant.Words.COLON;
+import static no.unit.nva.search2.constant.Words.COMMA;
+import static no.unit.nva.search2.constant.Words.DOT;
+import static no.unit.nva.search2.constant.Words.EQUAL;
+import static no.unit.nva.search2.constant.Words.KEYWORD;
+import static no.unit.nva.search2.constant.Words.ONE;
+import static no.unit.nva.search2.enums.ParameterKey.FieldOperator.GREATER_THAN_OR_EQUAL_TO;
+import static no.unit.nva.search2.enums.ParameterKey.FieldOperator.LESS_THAN;
+import static nva.commons.core.StringUtils.EMPTY_STRING;
+import static nva.commons.core.attempt.Try.attempt;
+import static nva.commons.core.paths.UriWrapper.fromUri;
+import java.net.URI;
+import java.net.URLDecoder;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import no.unit.nva.search2.constant.Defaults;
+import no.unit.nva.search2.constant.Words;
+import no.unit.nva.search2.enums.ParameterKey;
+import no.unit.nva.search2.enums.ParameterKey.ParamKind;
+import nva.commons.core.JacocoGenerated;
+import org.apache.lucene.search.join.ScoreMode;
+import org.opensearch.index.query.MatchQueryBuilder;
+import org.opensearch.index.query.MultiMatchQueryBuilder;
+import org.opensearch.index.query.MultiMatchQueryBuilder.Type;
+import org.opensearch.index.query.Operator;
+import org.opensearch.index.query.QueryBuilder;
+import org.opensearch.index.query.QueryBuilders;
+import org.opensearch.search.sort.SortOrder;
+
+public final class QueryTools<K extends Enum<K> & ParameterKey> {
+
+    public static Boolean valueToBoolean(String value) {
+        return ONE.equals(value) ? Boolean.TRUE : Boolean.valueOf(value);
+    }
+
+    public static String decodeUTF(String encoded) {
+        return URLDecoder.decode(encoded, StandardCharsets.UTF_8);
+    }
+
+    public static Collection<Entry<String, String>> queryToMapEntries(URI uri) {
+        return queryToMapEntries(uri.getQuery());
+    }
+
+    public static Collection<Entry<String, String>> queryToMapEntries(String query) {
+        return
+            nonNull(query)
+                ? Arrays.stream(query.split(AMPERSAND))
+                .map(s -> s.split(EQUAL))
```

Comment:
what is s?

Reply:
splitt by `& `and `=`

size=15&page=1 -> 'size=15',  'page=1' -> {{'size','15'}, { 'page','1'}}